core-java-modules/core-java-io2/src/main/java/com/baeldung/resource/MyResourceLoader.java

```diff
@@ -0,0 +1,42 @@
+package com.baeldung.resource;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.stream.Collectors;
+
+public class MyResourceLoader {
+
+    private void loadFileWithReader() throws IOException {
+
+        FileReader fileReader = new FileReader("src/main/resources/input.txt");
+        BufferedReader reader = new BufferedReader(fileReader);
+        String contents = reader.lines()
+            .collect(Collectors.joining(System.lineSeparator()));
+        reader.close();
+        System.out.println(contents);
+
+    }
+
+    private void loadFileAsResource() throws IOException {
+
+        InputStream inputStream = getClass().getResourceAsStream("/input.txt");
+        BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream));
+        String contents = reader.lines()
+            .collect(Collectors.joining(System.lineSeparator()));
+        System.out.println(contents);
+
+    }
+
+    public static void main(String[] args) throws IOException {
+
+        MyResourceLoader resourceLoader = new MyResourceLoader();
+
+        resourceLoader.loadFileWithReader();
+        resourceLoader.loadFileAsResource();
```

Comment:
The order might need to be reversed, so we can see that `loadFileAsResource` always works, where the other can fail (and stop the second being tried).